ursa_sharing/src/shamir.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 Hyperledger Ursa Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use super::{
+    error::{SharingError, SharingResult},
+    Field, Group, Polynomial,
+};
+use rand::{CryptoRng, RngCore};
+use std::{collections::BTreeSet, convert::TryFrom};
+use zeroize::Zeroize;
+
+/// A Shamir simple secret share
+/// provides no integrity checking
+#[derive(Debug, Zeroize)]
+#[zeroize(drop)]
+pub struct Share {
+    pub(crate) identifier: usize,
```

Comment:
You may want u32 rather than usize. 
usize is architecture dependent but below it is assumed to be 4 bytes. 
There's probably no benefit to 32 to 64 bit flexibility here since 32 bit will still give you 4 billion shares. (And frankly if most of earth has to simultaneously agree to something the aliens have already won.)